### DISCUSSION TITLE
如何从umi获取RequestOptions

### DISCUSSION THREAD

**Discussion by seafronthu** [umijs/umi#10522] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/50374650/218973048-671f670e-0cd7-4cf1-b775-be44874d9bb9.png)
以上是错误信息，umi中没有导出类型RequestOptions。我需要使用错误捕获方法二次封装request，所以需要里面的类型。有什么方式能够拿到吗

**Comment by seafronthu** [comment]
> ![image](https://user-images.githubusercontent.com/50374650/218973048-671f670e-0cd7-4cf1-b775-be44874d9bb9.png) 以上是错误信息，umi中没有导出类型RequestOptions。我需要使用错误捕获方法二次封装request，所以需要里面的类型。有什么方式能够拿到吗

![image](https://user-images.githubusercontent.com/50374650/218978676-00050936-08d9-4b87-90c6-aa8087df0c20.png)

**Comment by fz6m** [comment, marked as answer]
漏了，欢迎提 PR 在这里加一下：

https://github.com/umijs/umi/blob/e024b93e25c763921bb63ce1cb124cee35959f67/packages/plugins/src/request.ts#L314-L326

漏了这俩：

```ts
  RequestOptions,
  Request,
```

不过也可以通过其他方法取出来，比如：

```ts
import { RequestConfig } from '@umijs/max'

type RequestOptions = Parameters<Required<Required<RequestConfig>['errorConfig']>['errorHandler']>[1]
```
